File: FinalCode/blackScholes.py
import math
import numpy as np
from scipy.stats import norm

def blackScholes(tau, r, S_0, K, v, q, callOrPut):
    normalCDF = norm.cdf
    mult = 2 * (callOrPut.value - 1.5)

    d1 = (np.log(S_0 / K) + (r - q + (.5 * (v ** 2))) * tau) / (v * math.sqrt(tau))
    d2 = d1 - (v * math.sqrt(tau))
    return mult * (S_0 * (math.e ** (-q * tau)) * normalCDF(mult * d1) - (normalCDF(mult * d2) * K * math.e**(-r * tau)))

File: FinalCodeC++/blackScholes.cpp
#include "functions.h"

double blackScholes(double tau, double r, double S_0, double K, double v, double q, callPut callOrPut) {
    double mult = 2 * ((static_cast<int>(callOrPut)) - 1.5);
    
    double d1 = (log(S_0 / K) + (r - q + (0.5 * (pow(v, 2)))) * tau)/(v*sqrt(tau));
    double d2 = d1 - v * sqrt(tau);
    return mult * (S_0 * exp(-q * tau) * normalCDF(mult * d1) - normalCDF(mult * d2) * K * exp(-r * tau));
}


